### PR TITLE
Finite T Rydberg statistical t-tests

### PR DESCRIPTION
**Pull request by Atomyka**
Since we are not ready to merge the other files in Phil's original finite T pull request into master, this PR is intended for the Rydberg QMC statistical tests only.

### CODE REVIEW

File: lib/BloqadeQMC/test/rydberg.jl [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    for ii in 1:Δ_step
+        @show ii
+        H = Rydberg(lat, R_b, Ω, Δ[ii])
+        ts = BinaryThermalState(H, M)
+        d = Diagnostics()
+    
+        [mc_step_beta!(rng, ts, H, β, d, eq=true) for i in 1:EQ_MCS] #equilibration phase
+        
+        println("Equilibrated at this number of operators:")
+        @show mc_step_beta!(rng, ts, H, β, d, eq=true)
```
Comment (Roger-luo) [inline lib/BloqadeQMC/test/rydberg.jl:106]: can we remove the printings?

File: lib/BloqadeQMC/test/rydberg.jl [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        for i in 1:MCS # Monte Carlo Steps
+            ns[i] = mc_step_beta!(rng, ts, H, β, d, eq=false)
+        end
+
+        append!(energy_QMC_β3, mean_and_stderr(x -> -x/β, ns) + H.energy_shift)
+
+        @test abs(stdscore(energy_QMC_β3[ii], energy_ED[3,ii])) < THRESHOLD
+    end
+
+    scatter(Δ/2π, value.(energy_QMC_β3); yerror=uncertainty.(energy_QMC_β3), marker=:x, label="QMC")
```
Comment (Roger-luo) [inline lib/BloqadeQMC/test/rydberg.jl:158]: Why do we need plots in a test

**Review by Roger-luo — DISMISSED** [review]

**Comment by Roger-luo** [comment]
I think you forgot to include this test in runtests.jl that's why the test is passing but it actually not

File: lib/BloqadeQMC/test/rydberg.jl (hunk 1 above)
Comment (Atomyka) [reply lib/BloqadeQMC/test/rydberg.jl:106]: Ay ay sir :p

**Comment by Roger-luo** [comment]
you still need to include the test file so it actually get executed.

**Comment by Atomyka** [comment]
> you still need to include the test file so it actually get executed.

Yeah, I just ran out of data while traveling yesterday so couldn't include it. Updated now!

**Comment by Atomyka** [comment]
Seems like I need to add some binning or thinning after all - the seeds I was using worked on my laptop but not when Phil tried or here. (It's the very last high positive detuning frequency that is giving problems)

**Comment by Atomyka** [comment]
Fyi @Roger-luo we're still playing around with one chi-squared test to make it more rigorous. Will then merge!

**Review by weinbe58 — APPROVED** [review]
LGTM

**Comment by Atomyka** [comment]
Closing with working chi-squared test, including binned samples.

**Pull request merged**